### PR TITLE
feat: add login functions

### PR DESCRIPTION
**Pull request by amphro**
### What does this PR do?
Creates a login functions and modifies env list to separate heroku and functions.

### CODE REVIEW

File: src/commands/login/functions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+export default class FunctionsLogin extends SfCommand {
+  public static description = `login to a Salesforce account or enviornment
+
+  Login to https://login.salesforce.com in a browser. To login to different providers (Salesforce org, heroku, commerce cloud, mulesoft, etc) sepecify the domain or login url of the service. For example, '--instance-url=heroku.com'. 
+  `;
+
+  public static examples = [
+    'sf login --instance-url https://<mydomain>.my.salesforce.com',
+    'sf login --instance-url heroku.com',
```
Comment (RodEsp) [inline src/commands/login/functions.ts:23]:
```suggestion
	'sf login functions',
    'sf login functions --client-id XXXXXXXXXXXXXXX',
```

File: src/commands/login/functions.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { Flags } from '@oclif/core';
+import { AnyJson } from '@salesforce/ts-types';
+
+import SfCommand from '../../sf-command';
+import { login } from '../../utils';
+
+export default class FunctionsLogin extends SfCommand {
+  public static description = `login to a Salesforce account or enviornment
+
+  Login to https://login.salesforce.com in a browser. To login to different providers (Salesforce org, heroku, commerce cloud, mulesoft, etc) sepecify the domain or login url of the service. For example, '--instance-url=heroku.com'.
```
Comment (RodEsp) [inline src/commands/login/functions.ts:18]:
```suggestion
  public static description = `login to Salesforce functions environments

  Login to https://functions.salesforce.com in a browser. 
```

**Review by RodEsp — APPROVED** [review]
Looks good overall, just some help text edits. Feel free to merge with those updated.

As a side note the help text is kind of confusing. Seems weird to me for a working command, `sf login`, to have sub-commands `sf login functions` the way it's presented there.
image

File: src/commands/env/list.ts [hunk 3]
```diff
@@ -55,21 +55,20 @@ export default class EnvList extends SfCommand {
...
-      const type = name === 'heroku' ? 'compute' : 'org';
-      const accountEnvironments = account.environments.map((environmentsName) =>
-        this.convertEnvironment(environmentsName, {
+      const type = name === 'heroku' || name === 'functions' ? 'compute' : 'org';
+      const accountEnvironments = account.environments.map((environmentsName) => {
+        let context = 'sandbox';
+
+        if (name === 'heroku') context = 'heroku app';
+        if (name === 'functions') context = 'functions';
+        if (name === 'org' && environmentsName.includes('scratch')) context = 'scratch';
```
Comment (RodEsp) [inline src/commands/env/list.ts:64]: This is 100% just stylistic but I like how it reads better and that is makes intent more explicit. Feel free to ignore me if you disagree though :)

```suggestion
        let context;

        if (name === 'heroku') {
            context = 'heroku app';
      	} else if (name === 'functions') {
      	    context = 'functions';
      	} else if (name === 'org' && environmentsName.includes('scratch')) {
      	    context = 'scratch';
      	} else {
      	    context = 'sandbox';
      	}
```

Comment (amphro) [reply src/commands/env/list.ts:64]: I don't care, so sure. Although formatting is off on the suggestion so I will do in a followup.

File: src/commands/login/functions.ts (hunk 2 above)
Comment (amphro) [reply src/commands/login/functions.ts:18]: Ha ya, I forgot to do the help =)

I'll do this in the follow up with a few additions

**Pull request merged**